app/components/post_textbox/post_textbox_base.js

```diff
@@ -127,10 +129,15 @@ export default class PostTextBoxBase extends PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.channelId !== this.props.channelId || nextProps.rootId !== this.props.rootId) {
-            this.setState({value: nextProps.value});
+    static getDerivedStateFromProps(nextProps, state) {
+        if (nextProps.channelId !== state.channelId || nextProps.rootId !== state.rootId) {
```

Comment:
i wonder if [memoization](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization) wouldn't be better for this, as you could use the `channelId` and `rootId` as input and wouldn't have to track state

Reply:
>We’ve also seen derived state used to ensure an **expensive** value used in render is recomputed only when the inputs change. This technique is known as memoization.

Isn't the whole concept with memoization is in using expensive value. In our case it is a string

Reply:
yes, but it also abstract the state here so you wouldn't need to handle it, and hence my suggestion

Reply:
I don't think this is a place for memoization. We need to update the value of this textbox whenever these change because that's when the post draft changes. We're doing this to avoid updating the value at the wrong time which can mess with input, not to save computation time

Reply:
I wasn't looking to save computation time, but remove tracking state. Anyways, I'm 0/5 on this, so I'll change my review if you both agree on how it's done :)